ARTris/Localization/LocalizationViewController.swift

```diff
@@ -0,0 +1,129 @@
+    @IBOutlet weak var userPromptLabel: UILabel!
+    @IBOutlet weak var backBtn: UIButton!
+    @IBOutlet weak var sceneView: VirtualObjectARView! {
+        didSet {
+            let rotationHandler = #selector(LocalizationViewController.rotate(byReactingTo:))
+            rotationRecognizer = UIRotationGestureRecognizer(target: self, action: rotationHandler)
+            sceneView.addGestureRecognizer(rotationRecognizer)
+        }
+    }
+    @IBOutlet weak var nextBtn: UIButton! {
+        didSet {
+            nextBtn.isHidden = true
+        }
+    }
+    private lazy var gameViewController: GameViewController = {
+        let viewController = storyboard?.instantiateViewController(withIdentifier: "gameViewController") as! GameViewController
+        viewController.sceneView = sceneView
+        viewController.parentNode = shadowBinder
+        viewController.sessionId = sessionId
+        return viewController
+    }()
+    private var screenCentre: CGPoint {
+        let bounds = sceneView.bounds
+        return CGPoint(x: bounds.midX, y: bounds.midY)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        sceneView.session.delegate = self
+        sceneView.scene = SCNScene()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        let configuration = ARWorldTrackingConfiguration()
+        sceneView.session.run(configuration)
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        shadowBinder = SCNNode(geometry: nil)
+        shadow = GridShadow(w: 8, l: 8, parent: shadowBinder, size: 0.02, color: UIColor.green)
+        /*draw the shadow grid but do not add its parent node to the ARSCNView yet*/
+        shadow.draw()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+    }
+    
+    func session(_ session: ARSession, didUpdate frame: ARFrame) {
+        if let trackingState = sceneView.session.currentFrame?.camera.trackingState {
+            updateTrackingStateLabel(trackingstate: trackingState)
+        }
+        var objectPosition: float3?
+        let shadowBinderPosition = float3(shadowBinder.position)
+        
+        /*If the position of the shadow binder is the default vector: (0,0,0), keep the objectPosition as nil*/
+        if shadowBinderPosition != float3(0,0,0) {
+            objectPosition = shadowBinderPosition
+        }
+        
+        /*Perform a hit test at the screen centre*/
+        guard let (worldPosition,_,_) =  sceneView.worldPosition(fromScreenPosition: screenCentre, objectPosition: objectPosition) else { return }
+        /*update the position of the shadow binder*/
+        shadowBinder.position = SCNVector3(worldPosition)
+        
+        /*Once the object has been placed, the "next" button should appear to allow the user to transition to gameplay*/
+        if objectPosition != nil {
```

Comment:
can you define a boolean variable to capture this condition? You can name it something like `gridHasBeenPlaced`